packages/js/src/indexables-page/components/not-enough-analysed-content.js

```diff
@@ -0,0 +1,96 @@
+import PropTypes from "prop-types";
+
+import { __, sprintf } from "@wordpress/i18n";
+
+import SimpleCard from "./simple-card";
+import { useState, useCallback } from "@wordpress/element";
+import { Button, Link } from "@yoast/ui-library";
+import { IndexableLinkCount } from "../indexables-page";
+
+/**
+ * Renders the four indexable tables.
+ *
+ * @returns {WPElement} A div containing the empty state page.
+ */
+const NotEnoughAnalysedContent = ( { indexablesList } ) => {
+	const [ step, setStep ] = useState( 5 );
+	const [ numberOfVisibleIndexables, setNumberOfVisibleIndexables ] = useState( step );
+
+	const handleShowMore = useCallback( ( e ) => {
+		e.preventDefault();
+		const number = numberOfVisibleIndexables + step > indexablesList.length ? indexablesList.length : numberOfVisibleIndexables + step;
+		const newStep = indexablesList.length - number > step ? step :  indexablesList.length - number;
+		setNumberOfVisibleIndexables( number );
+		setStep( newStep );
+	}, [ numberOfVisibleIndexables, setNumberOfVisibleIndexables ] );
+	return <div className="yst-max-w-full yst-mt-6">
+		<div
+			id="start-writing-content"
+			className="yst-max-w-7xl yst-grid yst-grid-cols-1 2xl:yst-grid-cols-2 2xl:yst-grid-rows-2 2xl:yst-grid-flow-row 2xl:yst-auto-rows-fr yst-gap-6"
+		>
+			<SimpleCard
+				title={
+					/* translators: %1$s expands to the number of posts without a focus keyphrase */
+					sprintf(
+						__(
+							"Posts without a focus keyphrase (%1$s)",
+							"wordpress-seo"
+						),
+						indexablesList.length
+					)
+				}
+			>
+				<ul className="yst-divide-y yst-divide-gray-200">
+					{ indexablesList.slice( 0, numberOfVisibleIndexables ).map(
```

Comment:
perhaps it's slightly more preferable to keep a list of the visible indexables in the state, rather than recalculating each render. The handleShowMore could then set the state by slicing.

Not sure it's worth the effort though.